### DISCUSSION TITLE
Kubeapps URL

### DISCUSSION THREAD

**Discussion by tjanitsc** [kubeapps/kubeapps#4229] [Q&A] [answered]
How can I change the Access Url of the Kubeapps deployment. I want access the Kubeapp from outside the cluster. So the 127.0.0.1 isnt accsessable in my network.

**Comment by antgamdia** [comment, marked as answer]
You would need to expose the Kubernetes service externally, more info: https://medium.com/google-cloud/kubernetes-nodeport-vs-loadbalancer-vs-ingress-when-should-i-use-what-922f010849e0.

More specifically, regarding Kubeapps, have a look at this FAQ section: https://github.com/kubeapps/kubeapps/blob/main/chart/kubeapps/README.md#how-to-configure-kubeapps-with-ingress

**Comment by tjanitsc** [comment]
its behind a nginx loadbalancer
But what I have to change in this environment. Which configuration file?

**Reply by antgamdia** [reply]
I guess you'll need to install Kubeapps with the ingress feature on. It will create an Ingress rule in your cluster, so that you can access it externally. For example, for pointing to `example.com` you'll need to run:

```bash
helm install kubeapps bitnami/kubeapps \
  --namespace kubeapps \
  --set ingress.enabled=true \
  --set ingress.hostname=example.com \
  --set ingress.annotations."kubernetes\.io/ingress\.class"=nginx # or your preferred ingress controller
```

**Comment by tjanitsc** [comment]
how long does it take to install kubeapps?

**Reply by antgamdia** [reply]
From my experience, it shouldn't take more than 1-5 mins

**Comment by tjanitsc** [comment]
i think it has a problem to pull the data

helm install kubeapps c:\kubeapps --namespace kubeapps --set ingress.enabled=true --set ingr
ess.hostname=cluster2ingress.berlab.de --set ingress.path=/kubeapps --set ingress.annotations."kubernetes\.io/ingress\.class"=
essential-nginx-ingress-ingress-nginx-controller
NAME: kubeapps
LAST DEPLOYED: Thu Feb  3 11:13:03 2022
NAMESPACE: kubeapps
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
CHART NAME: kubeapps
CHART VERSION: 7.7.2
APP VERSION: 2.4.2** Please be patient while the chart is being deployed **

Tip:

  Watch the deployment status using the command: kubectl get pods -w --namespace kubeapps

Kubeapps can be accessed via port 80 on the following DNS name from within your cluster:

   kubeapps.kubeapps.svc.cluster.local

To access Kubeapps from outside your K8s cluster, follow the steps below:

1. Get the Kubeapps URL and associate Kubeapps hostname to your cluster external IP:

   export CLUSTER_IP=$(minikube ip) # On Minikube. Use: `kubectl cluster-info` on others K8s clusters
   echo "Kubeapps URL: http://cluster2ingress.berlab.de/"
   echo "$CLUSTER_IP  cluster2ingress.berlab.de" | sudo tee -a /etc/hosts

2. Open a browser and access Kubeapps using the obtained URL.

##########################################################################################################
### WARNING: You did not provide a value for the postgresqlPassword so one has been generated randomly ###
##########################################################################################################

kubectl get pods -w --namespace kubeapps
NAME                                                          READY   STATUS              RESTARTS   AGE
kubeapps-75ccfd496b-64sqj                                     0/1     ContainerCreating   0          38s
kubeapps-75ccfd496b-qg6dh                                     0/1     ContainerCreating   0          38s
kubeapps-internal-apprepository-controller-7f5c4d7db6-pmbbv   0/1     ImagePullBackOff    0          38s
kubeapps-internal-dashboard-6fd9f465f8-g4895                  0/1     ImagePullBackOff    0          38s
kubeapps-internal-dashboard-6fd9f465f8-kz5v6                  0/1     ErrImagePull        0          38s
kubeapps-internal-kubeappsapis-6c88c5fbc4-p5mvv               0/1     ErrImagePull        0          38s
kubeapps-internal-kubeappsapis-6c88c5fbc4-rcdg8               0/1     ContainerCreating   0          38s
kubeapps-internal-kubeops-665c99759d-ddqpm                    0/1     ContainerCreating   0          38s
kubeapps-internal-kubeops-665c99759d-dvfr5                    0/1     ContainerCreating   0          38s
kubeapps-postgresql-primary-0                                 0/1     ContainerCreating   0          38s
kubeapps-postgresql-read-0                                    0/1     ContainerCreating   0          38s
kubeapps-internal-dashboard-6fd9f465f8-kz5v6                  0/1     ImagePullBackOff    0          43s
kubeapps-internal-kubeappsapis-6c88c5fbc4-p5mvv               0/1     ImagePullBackOff    0          45s
kubeapps-postgresql-primary-0                                 0/1     ErrImagePull        0          47s
kubeapps-75ccfd496b-64sqj                                     0/1     ErrImagePull        0          47s
kubeapps-postgresql-primary-0                                 0/1     ImagePullBackOff    0          59s
kubeapps-75ccfd496b-64sqj                                     0/1     ImagePullBackOff    0          61s
kubeapps-internal-kubeops-665c99759d-ddqpm                    0/1     ErrImagePull        0          62s
kubeapps-internal-kubeappsapis-6c88c5fbc4-rcdg8               0/1     ErrImagePull        0          62s
PS C:\Users\administrator.BERLAB> kubectl get pods -w --namespace kubeapps
NAME                                                          READY   STATUS             RESTARTS   AGE
kubeapps-75ccfd496b-64sqj                                     0/1     ImagePullBackOff   0          15m
kubeapps-75ccfd496b-qg6dh                                     0/1     ImagePullBackOff   0          15m
kubeapps-internal-apprepository-controller-7f5c4d7db6-pmbbv   0/1     ImagePullBackOff   0          15m
kubeapps-internal-dashboard-6fd9f465f8-g4895                  0/1     ImagePullBackOff   0          15m
kubeapps-internal-dashboard-6fd9f465f8-kz5v6                  0/1     ImagePullBackOff   0          15m
kubeapps-internal-kubeappsapis-6c88c5fbc4-p5mvv               0/1     ImagePullBackOff   0          15m
kubeapps-internal-kubeappsapis-6c88c5fbc4-rcdg8               0/1     ImagePullBackOff   0          15m
kubeapps-internal-kubeops-665c99759d-ddqpm                    0/1     ImagePullBackOff   0          15m
kubeapps-internal-kubeops-665c99759d-dvfr5                    0/1     ErrImagePull       0          15m
kubeapps-postgresql-primary-0                                 0/1     ImagePullBackOff   0          15m
kubeapps-postgresql-read-0                                    0/1     ImagePullBackOff   0          15m
kubeapps-internal-kubeops-665c99759d-dvfr5                    0/1     ImagePullBackOff   0          15m
kubeapps-internal-kubeops-665c99759d-ddqpm                    0/1     ErrImagePull       0          15m
kubeapps-internal-kubeops-665c99759d-ddqpm                    0/1     ImagePullBackOff   0          16m
kubeapps-postgresql-read-0                                    0/1     ErrImagePull       0          16m

**Reply by tjanitsc** [reply]
where can i download a deployable zip?

**Reply by antgamdia** [reply]
Kubeapps is officially distributed as a Helm chart in the [Bitnami Application Catalog](https://bitnami.com/application-catalog), so if want to get a zip, you just need to go to [their charts repo](https://github.com/bitnami/charts) and export it as a zip from GitHub, that is: https://github.com/bitnami/charts/archive/refs/heads/master.zip

Then, just go to bitnami/kubeapps and there you'll find the official Kubeapps chart.

**Reply by tjanitsc** [reply]
ok, but what are the right steps?
1. install the chart
2. update this like: helm dep update C:\kubeapps

**Reply by tjanitsc** [reply]
I dont find a right process step by step...

**Reply by antgamdia** [reply]
We don't have any specific guide for installing Kubeapps if, in your environment, Helm is not fully able to retrieve the assets.
The usual [getting started](https://github.com/kubeapps/kubeapps/blob/main/docs/user/getting-started.md) should already cover how to install and use Kubeapps.

If you are working in an air-gapped environment, you also may find [this another guide](https://github.com/kubeapps/kubeapps/blob/main/docs/user/offline-installation.md) useful.
